libdevcore/db.h

```diff
@@ -70,14 +70,16 @@ class DatabaseFace
     virtual void forEach(std::function<bool(Slice, Slice)> f) const = 0;
 };
 
-DEV_SIMPLE_EXCEPTION(FailedToOpenDB);
-DEV_SIMPLE_EXCEPTION(FailedInsertInDB);
-DEV_SIMPLE_EXCEPTION(FailedLookupInDB);
-DEV_SIMPLE_EXCEPTION(FailedDeleteInDB);
-DEV_SIMPLE_EXCEPTION(FailedCommitInDB);
-DEV_SIMPLE_EXCEPTION(FailedIterateDB);
-DEV_SIMPLE_EXCEPTION(DBCorruption);
-DEV_SIMPLE_EXCEPTION(DBIOError);
+DEV_SIMPLE_EXCEPTION(DatabaseError);
+struct DBCorruption : virtual DatabaseError
+{
+};
+struct IOError : virtual DatabaseError
+{
+};
+struct NotFound : virtual DatabaseError
```

Comment:
By following Java convention: Errors are fatal, Exceptions not. So I'd suggest renaming it to:

- DatabaseError -> DatabaseException (because NotFound is part of it)
- DBCorruption -> (DB)CorruptionError

Reply:
Hm, this would break "is-a" relationship between `CorruptionError` and `DatabaseException`